tokio-trace/tests/macros.rs

```diff
@@ -8,95 +10,258 @@ extern crate tokio_trace;
-    span!(target: "foo_events", level: tokio_trace::Level::DEBUG, "bar",);
-    span!(level: tokio_trace::Level::DEBUG, "foo", bar = 2, baz = 3);
-    span!(level: tokio_trace::Level::DEBUG, "foo", bar = 2, baz = 4,);
-    span!(level: tokio_trace::Level::DEBUG, "foo");
-    span!(level: tokio_trace::Level::DEBUG, "bar",);
-    span!("foo", bar = 2, baz = 3);
-    span!("foo", bar = 2, baz = 4,);
-    span!("foo");
-    span!("bar",);
+    span!(Level::DEBUG, target: "foo_events", "foo", bar = 2, baz = 3);
+    span!(Level::DEBUG, target: "foo_events", "foo", bar = 2, baz = 4,);
+    span!(Level::DEBUG, target: "foo_events", "foo");
+    span!(Level::DEBUG, target: "foo_events", "bar",);
+    span!(Level::DEBUG, "foo", bar = 2, baz = 3);
+    span!(Level::DEBUG, "foo", bar = 2, baz = 4,);
+    span!(Level::TRACE, "foo", bar = 2, baz = 3);
+    span!(Level::TRACE, "foo", bar = 2, baz = 4,);
+    span!(Level::TRACE, "foo");
+    span!(Level::TRACE, "bar",);
+}
+
+#[test]
+fn trace_span() {
+    trace_span!(target: "foo_events", "foo", bar = 2, baz = 3);
+    trace_span!(target: "foo_events", "foo", bar = 2, baz = 4,);
+    trace_span!(target: "foo_events", "foo");
+    trace_span!(target: "foo_events", "bar",);
+    trace_span!("foo", bar = 2, baz = 3);
+    trace_span!("foo", bar = 2, baz = 4,);
+    trace_span!("bar");
+    trace_span!("bar",);
+}
+
+#[test]
+fn debug_span() {
+    debug_span!(target: "foo_events", "foo", bar = 2, baz = 3);
+    debug_span!(target: "foo_events", "foo", bar = 2, baz = 4,);
+    debug_span!(target: "foo_events", "foo");
+    debug_span!(target: "foo_events", "bar",);
+    debug_span!("foo", bar = 2, baz = 3);
+    debug_span!("foo", bar = 2, baz = 4,);
+    debug_span!("bar");
+    debug_span!("bar",);
+}
+
+#[test]
+fn info_span() {
+    info_span!(target: "foo_events", "foo", bar = 2, baz = 3);
+    info_span!(target: "foo_events", "foo", bar = 2, baz = 4,);
+    info_span!(target: "foo_events", "foo");
+    info_span!(target: "foo_events", "bar",);
+    info_span!("foo", bar = 2, baz = 3);
+    info_span!("foo", bar = 2, baz = 4,);
+    info_span!("bar");
+    info_span!("bar",);
+}
+
+#[test]
+fn warn_span() {
+    warn_span!(target: "foo_events", "foo", bar = 2, baz = 3);
+    warn_span!(target: "foo_events", "foo", bar = 2, baz = 4,);
+    warn_span!(target: "foo_events", "foo");
+    warn_span!(target: "foo_events", "bar",);
+    warn_span!("foo", bar = 2, baz = 3);
+    warn_span!("foo", bar = 2, baz = 4,);
+    warn_span!("bar");
+    warn_span!("bar",);
+}
+
+#[test]
+fn error_span() {
+    error_span!(target: "foo_events", "foo", bar = 2, baz = 3);
+    error_span!(target: "foo_events", "foo", bar = 2, baz = 4,);
+    error_span!(target: "foo_events", "foo");
+    error_span!(target: "foo_events", "bar",);
+    error_span!("foo", bar = 2, baz = 3);
+    error_span!("foo", bar = 2, baz = 4,);
+    error_span!("bar");
+    error_span!("bar",);
```

Comment:
Since these tests have gotten so repetitive, part of me wonders if we should be generating them with a macro. But since the purpose of these tests is to catch any macro compilation errors, that's probably a mistake...